### ISSUE TITLE
Runs multiple times per second even after setting schedule and repeat

### ISSUE THREAD

**Issue by shivanish** [arteria/django-background-tasks#200]
I  have added this code to my models. I want the background task to run only once per minute. So i gave the value for `schedule` as 60 (seconds) and `repeat` = 60 (seconds).

```
from background_task import background
@background(schedule=60)
def update_queue_status():
    print(datetime.now().time())
    ................
    ...............
            
update_queue_status(repeat=60)
```

But when i run `python manage.py process_tasks`, it runs the background task multiple times per second-
```
(qms_env) (base) C:\Users\shisharma\Documents\futuredev\QMS>python manage.py process_tasks
11:51:06.860161
11:51:06.911026
11:51:06.945932
11:51:07.022755
11:51:07.055640
11:51:07.071596
11:51:07.145399
11:51:07.165345
11:51:07.261117
11:51:07.276049
11:51:07.367802
11:51:07.461586
11:51:07.535353
11:51:07.630100
11:51:07.715872
11:51:07.784686
11:51:07.816629
11:51:07.825579
11:51:07.918329
11:51:07.977172
11:51:08.027037
11:51:08.112807
11:51:08.224510
11:51:08.280358
11:51:08.345213
11:51:08.442924
11:51:08.548641
11:51:08.587536
11:51:08.614464
11:51:08.632415
11:51:08.705221
11:51:08.751128
11:51:08.855819
11:51:08.901694
11:51:08.937628
11:51:08.978518
11:51:09.064289
11:51:09.155046
11:51:09.225828
11:51:08.941590
11:51:09.108143
11:51:08.927654
11:51:09.013395
11:51:09.197934
11:51:59.796843
11:51:59.824775
11:51:59.930155
11:52:00.026355
11:52:00.041316
11:52:00.113123
11:52:00.155011
```

Is there something wrong with what i am doing. 

Thank you
Shivani

**Comment by shivanish** [comment]
@invi84 @philippeowagner @jvamvas  Any suggestions, please ?

**Comment by sanket0354** [comment]
+1 Would like to know the reason. I have a similar problem.

**Comment by Nebucatnetzer** [comment]
I have the same problem. I want to write sensor data to the database every 10 seconds or so. Currently it does it like 10 times or more per run.

**Comment by philippeowagner** [comment]
What versions of Python and Django are used in this case?

**Comment by Nebucatnetzer** [comment]
For me its Python 3.7 with Django 2.2.7

**Comment by tehfink** [comment]
Seeing this as well on Django 3.0.1, Python 3.8.0, and django-background-tasks 1.2.5.

**Comment by aditya88** [comment]
I see this happening with my code too. Did you get a chance to see if there are multiple tasks with the same hash in background_task table? BTW I am using async IO code in my tasks(not sure if it has anything to do with this).

**Comment by MohitThorat** [comment]
I am also facing this problem in Django 3.0 and Python 3.7

**Comment by kyamasam** [comment]
This is happening to me too on Django 3.0.2 and backgound_task 1.2.5

**Comment by uros94** [comment]
I tried removing old tasks from the database. Didn't help...

**Comment by jan-pachon** [comment]
I am having the same issue. What is the solution?

**Comment by stoicaandrei** [comment]
Managed to solve it by doing
```
from background_task.models import Task
Task.objects.all().delete()
```
It seems that the problem appears after changing the `repeat` argument. The old tasks keep executing alongside the new ones.

**Comment by techon** [comment]
As noted above, deleting all tasks works. This is because a new task is created each time the process_tasks command is run.
So if you've stopped/started/stopped/started process_tasks for a repeating task, you will end up with multiples of the same task.

I've further added to the above response to only delete the tasks created by the current app.

    from background_task.models import Task
    """
    Tasks, when stored in database, are prepended with the app name. 
    This will clear tasks for the current app only. 
    """
    Task.objects.filter(task_name__startswith=__name__).delete()

Alternatively, I moved on to a different solution. I expect non-repeating tasks to get queued up each time the script is run (This is current behavior), I do NOT want repeating tasks to create a new repeating task. 
To address this, I've cloned background_task and added the following to background_task.tasks.DBTaskRunner.schedule. This will continue default behavior for non-repeating task and create a new task. However, when a repeating task is specified, the pre-existing task entry will be updated. -- You will have to delete any related tasks (in case of multiple Task entries) before transitioning to the below method.

I added the following snippet to background_task.tasks.DBTaskRunner.schedule right after new_task creation (task = Task.objects.new_task(...)

    '''
    Default background_task behavior is to queue a new task whether repeating or not. 
    Instead, if repeat is specified, reschedule existing task entry waiting to be run.
    This prevents queue from getting bloated with duplicate repeating schedules.
    '''
    if not action and repeat:
        action = TaskSchedule.SCHEDULE
    else:
        action = TaskSchedule.RESCHEDULE_EXISTING

Depending on how you are using background_task, you may want to consider further adding to the above modification to compare repeat_until against run_at to determine if there is no overlap and if indeed a "duplicate" Task in the queue is desired.

**Comment by cmosquer** [comment]
For me the solution was using one of django-background-task available constants: 

"The following constants are provided: Task.NEVER (default), Task.HOURLY, Task.DAILY, Task.WEEKLY, Task.EVERY_2_WEEKS, Task.EVERY_4_WEEKS."

Using these constants for the 'repeat' parameter when calling the task, solved the issue for me.
